packages/rest/lib/resource.cadl

```diff
@@ -110,3 +110,89 @@ interface ResourceInstanceOperations<TResource, TError> mixes ResourceRead<TReso
 interface ResourceCollectionOperations<TResource, TError> mixes ResourceCreate<TResource, TError>, ResourceList<TResource, TError> {}
 
 interface ResourceOperations<TResource, TError> mixes ResourceInstanceOperations<TResource, TError>, ResourceCollectionOperations<TResource, TError> {}
+
+interface SingletonResourceRead<TSingleton, TResource, TError> {
```

Comment:
Can we add docs on the interfaces of rest(comment docs not @doc) on what each interface does. 
I think from someone coming there the first time it can be quite intimindating with all those interface to understand the final result

Reply:
I think it would be very nice to be able to get that documentation as well in the IDE latter.

Reply:
Will do!